### ISSUE TITLE
Limitations

### ISSUE THREAD

**Issue by ttuegel** [kframework/kore#1631]
This issue is to track the most significant known limitations of the prover compared to the Java version.

- #1183: Matching to iterate over Map and Set
  The Haskell backend does not support function rules to iterate over Map and Set collections. Why: Iterating over Map and Set collections is not deterministic with respect to the order of items and confluence is not guaranteed. Workaround: For concrete Map and Set terms, use `Set2List` and iterate over the list instead, or use rewrite rules.
- #1557: Attributes to replace #isConcrete
  The Haskell backend does not support `KREFLECTION`. Why: There is no sound way to translate the functions in `KREFLECTION` to Kore symbols.
- #1416 Include the side condition when evaluating functions in the side condition
  Side conditions are not used to evaluate themselves. In particular, this hampers evaluation of functions in the top-level conditions.

**Comment by ttuegel** [comment]
> What if Set has symbolic elements? Will the workaround work?

No. The workaround avoids inconsistency because the hooked `Set2List` function is held unevaluated until its argument is concrete, at which point we can arbitrarily, but consistently, order the elements.

**Comment by nishantjr** [comment]
Should #1817  also be here?

**Comment by ttuegel** [comment]
In my opinion, no. That is a bug that affects the soundness of the prover, but this issue tracks limitations that affect the completeness of the prover.
